test/http-test.js

```diff
@@ -344,6 +344,65 @@ describe('HTTP', function() {
     assert.equal(filterInfo.entries, 1);
   });
 
+  it('should generate 10 blocks from RPC call', async () => {
+    const blocks = await nclient.execute('generatetoaddress', [10, addr.toString(network)]);
+    assert.strictEqual(blocks.length, 10);
+  });
+
+  // depends on the previous test to generate blocks
+  it('should fetch block header by height', async () => {
+    // fetch corresponding header and block
+    const height = 7;
+    const header = await nclient.get(`/header/${height}`);
+    assert.equal(header.height, height);
+
+    const properties = [
+      'hash', 'version', 'prevBlock',
+      'merkleRoot', 'time', 'bits',
+      'nonce', 'height', 'chainwork'
+    ];
+
+    for (const property of properties)
+      assert(property in header);
+
+    const block = await nclient.getBlock(height);
+
+    assert.equal(block.hash, header.hash);
+    assert.equal(block.height, header.height);
+    assert.equal(block.version, header.version);
+    assert.equal(block.prevBlock, header.prevBlock);
+    assert.equal(block.merkleRoot, header.merkleRoot);
+    assert.equal(block.time, header.time);
+    assert.equal(block.bits, header.bits);
+    assert.equal(block.nonce, header.nonce);
+  });
+
+  it('should fetch null for block header that does not exist', async () => {
+    // many blocks in the future
+    const header = await nclient.get(`/header/${40000}`);
```

Comment:
Curious why is `40000` in braces?
This could be foolproof by getting chain tip height and adding any value > 1
But no change needed really.